Compiling gtsam using aarch64-linux-android clang++ failed

## Description

Compiling gtsam using aarch64-linux-andrord clang toolchain

## Additional information
```
/workplace/gtsam/gtsam/navigation/NavState.cpp:109:9: error: cannot decompose non-public member 'R_' of 'gtsam::NavState'
    auto [nRb, n_t, n_v] = (*this);
/workplace/gtsam/gtsam/navigation/NavState.h:39:8: note: declared private here
    Rot3 R_; ///< Rotation nRb, rotates points/velocities in body to points/velocities in nav
```
## TODO something
   Try to change private to public to compile normally

--- a/gtsam/navigation/NavState.cpp
+++ b/gtsam/navigation/NavState.cpp
@@ -106,7 +106,8 @@ bool NavState::equals(const NavState& other, double tol) const {
 //------------------------------------------------------------------------------
 NavState NavState::retract(const Vector9& xi, //
     OptionalJacobian<9, 9> H1, OptionalJacobian<9, 9> H2) const {
-  auto [nRb, n_t, n_v] = (*this);
+  Rot3 nRb = R_;
+  Point3 n_t = t_, n_v = v_;
   Matrix3 D_bRc_xi, D_R_nRb, D_t_nRb, D_v_nRb;
   const Rot3 bRc = Rot3::Expmap(dR(xi), H2 ? &D_bRc_xi : 0);
   const Rot3 nRc = nRb.compose(bRc, H1 ? &D_R_nRb : 0);

